python/plugins/processing/ProcessingPlugin.py

```diff
@@ -219,6 +221,14 @@ def initGui(self):
         self.resultsAction.setText(self.tr('&Results Viewer'))
         self.iface.registerMainWindowAction(self.resultsAction, 'Ctrl+Alt+R')
         self.menu.addAction(self.resultsAction)
+        self.toolbox.processingToolbar.addAction(self.resultsAction)
+
+        self.optionsAction = QAction(
+            QgsApplication.getThemeIcon("/processingAlgorithm.svg"),
+            self.tr('Options'), self.iface.mainWindow())
+        self.optionsAction.setObjectName('optionsAction')
+        self.optionsAction.triggered.connect(self.openProcessingOptions)
+        self.toolbox.processingToolbar.addAction(self.optionsAction)
```

Comment:
You'll need a call somewhere here for:

     self.toolbox.processingToolbar.setIconSize(self.iface.iconSize(dockedToolbar=True))

Otherwise hidpi users will come and attack you with pitchforks

Reply:
![giphy](https://user-images.githubusercontent.com/1374682/35304607-5e2ac98c-0096-11e8-8600-19962e01e58c.gif)